GrpcInterface/RiaGrpcGridGeometryExtractionService.cpp

```diff
@@ -0,0 +1,718 @@
+    {
+        filtColl->removeFilter( filter );
+    }
+
+    // Add range filter object
+    const int sliceDirection = -1;
+    const int gridIndex      = 0;
+    RimCellRangeFilter* cellRangeFilter = filtColl->addNewCellRangeFilter( view->eclipseCase(), gridIndex, sliceDirection );
+
+    // Eclipse indexing, first index is 1
+    // Dimension filter is disabled if both min and max is -1 for respective dimension. I.e. all cells for
+    // respective dimension are included in the filter.
+    const bool isIDimensionFilterDisabled = filter.imin() == -1 && filter.imax() == -1;
+    int        startIndexI                = filter.imin() + 1;
+    int        cellCountI                 = filter.imax() - filter.imin() + 1;
+    if ( isIDimensionFilterDisabled )
+    {
+        startIndexI = 1;
+        cellCountI  = view->mainGrid()->cellCountI();
+    }
+
+    const bool isJDimensionFilterDisabled = filter.jmin() == -1 && filter.jmax() == -1;
+    int        startIndexJ                = filter.jmin() + 1;
+    int        cellCountJ                 = filter.jmax() - filter.jmin() + 1;
+    if ( isJDimensionFilterDisabled )
+    {
+        startIndexJ = 1;
+        cellCountJ  = view->mainGrid()->cellCountJ();
+    }
+
+    const bool isKDimensionFilterDisabled = filter.kmin() == -1 && filter.kmax() == -1;
+    int        startIndexK                = filter.kmin() + 1;
+    int        cellCountK                 = filter.kmax() - filter.kmin() + 1;
+    if ( isKDimensionFilterDisabled )
+    {
+        startIndexK = 1;
+        cellCountK  = view->mainGrid()->cellCountK();
+    }
+
+    // Apply ijk-filter values to range filter object
+    cellRangeFilter->startIndexI = startIndexI;
+    cellRangeFilter->startIndexJ = startIndexJ;
+    cellRangeFilter->startIndexK = startIndexK;
+    cellRangeFilter->cellCountI  = cellCountI;
+    cellRangeFilter->cellCountJ  = cellCountJ;
+    cellRangeFilter->cellCountK  = cellCountK;
+
+    return grpc::Status::OK;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+grpc::Status RiaGrpcGridGeometryExtractionService::initializeGridGeometryGeneratorWithEclipseViewCellVisibility(
+    cvf::StructGridGeometryGenerator& surfaceGeometryGenerator,
+    cvf::StructGridGeometryGenerator& faultGeometryGenerator,
+    RimEclipseView*                   view )
+{
+    if ( view == nullptr )
+    {
+        return grpc::Status( grpc::StatusCode::NOT_FOUND, "Uninitialized eclipse view provided" );
+    }
+
+    auto* mainGrid = view->mainGrid();
+    if ( mainGrid == nullptr )
+    {
+        return grpc::Status( grpc::StatusCode::NOT_FOUND, "No main grid found for eclipse view" );
+    }
+
+    // Cell visibilities
+    const int firstTimeStep    = 0;
+    auto*     cellVisibilities = new cvf::UByteArray( mainGrid->cellCount() );
```

Comment:
`cellVisibilities` must be put into a cvf::ref member in `RiaGrpcGridGeometryExtractionService`